Subject: [PATCH] fixed cors issue
---
Index: blogs-analyzer-ui/src/app/app.module.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/app.module.ts b/blogs-analyzer-ui/src/app/app.module.ts
--- a/blogs-analyzer-ui/src/app/app.module.ts	(revision 346accf1d385fae931da0747f821591d0c770d3f)
+++ b/blogs-analyzer-ui/src/app/app.module.ts	(date 1718543308123)
@@ -1,24 +1,41 @@
-import { NgModule } from '@angular/core';
-import { BrowserModule } from '@angular/platform-browser';
+import {NgModule} from '@angular/core';
+import {BrowserModule} from '@angular/platform-browser';
 
-import { AppRoutingModule } from './app-routing.module';
-import { AppComponent } from './app.component';
-import { DashboardComponent } from './dashboard/dashboard.component';
-import { HeaderComponent } from './dashboard/components/header/header.component';
+import {AppRoutingModule} from './app-routing.module';
+import {AppComponent} from './app.component';
+import {DashboardComponent} from './dashboard/dashboard.component';
+import {HeaderComponent} from './dashboard/components/header/header.component';
 import {MaterialModule} from "./shared/module/material.module";
+import {HttpClientModule} from "@angular/common/http";
+import {FormsModule} from "@angular/forms";
+import {TabularViewComponent} from './dashboard/components/tabular-view/tabular-view.component';
+import {AgGridModule} from "ag-grid-angular";
+import {MatCardModule} from "@angular/material/card";
+import {BrowserAnimationsModule} from "@angular/platform-browser/animations";
+import {MatIconModule} from "@angular/material/icon";
+import {HomeComponent} from './dashboard/components/home/home.component';
 
 @NgModule({
-  declarations: [
-    AppComponent,
-    DashboardComponent,
-    HeaderComponent
-  ],
-  imports: [
-    BrowserModule,
-    AppRoutingModule,
-    MaterialModule
-  ],
-  providers: [],
-  bootstrap: [AppComponent]
+    declarations: [
+        AppComponent,
+        DashboardComponent,
+        HeaderComponent,
+        TabularViewComponent,
+        HomeComponent
+    ],
+    imports: [
+        BrowserModule,
+        BrowserAnimationsModule,
+        AppRoutingModule,
+        MaterialModule,
+        HttpClientModule,
+        AgGridModule,
+        FormsModule,
+        MatCardModule,
+        MatIconModule
+    ],
+    providers: [],
+    bootstrap: [AppComponent]
 })
-export class AppModule { }
+export class AppModule {
+}
Index: blogs-analyzer-ui/src/app/dashboard/services/blog.service.spec.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/services/blog.service.spec.ts b/blogs-analyzer-ui/src/app/dashboard/services/blog.service.spec.ts
new file mode 100644
--- /dev/null	(date 1718366925849)
+++ b/blogs-analyzer-ui/src/app/dashboard/services/blog.service.spec.ts	(date 1718366925849)
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
+
+import { BlogService } from './blog.service';
+
+describe('BlogService', () => {
+  let service: BlogService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(BlogService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+});
Index: blogs-analyzer-ui/src/app/dashboard/services/blog.service.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/services/blog.service.ts b/blogs-analyzer-ui/src/app/dashboard/services/blog.service.ts
new file mode 100644
--- /dev/null	(date 1718539370475)
+++ b/blogs-analyzer-ui/src/app/dashboard/services/blog.service.ts	(date 1718539370475)
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import {map, Observable} from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BlogService {
+  private baseUrl = 'http://localhost:8080/api/wordpress';
+
+
+  constructor(private http: HttpClient) { }
+
+  searchPostsByTitle(title: string): Observable<any> {
+    const url = `${this.baseUrl}/posts-by-title?title=${encodeURIComponent(title)}`;
+    return this.http.get(url, { responseType: 'text' });
+  }
+
+  getAllPosts(page: number, pageSize: number): Observable<any> {
+    const url = `${this.baseUrl}?page=${page}&size=${pageSize}`;
+    return this.http.get(url).pipe(
+        map((response: any) => {
+          return {
+            posts: response.posts,
+            totalPages: response.totalPages
+          };
+        })
+    );
+  }
+
+  getPostById(id: number): Observable<any> {
+    const url = `${this.baseUrl}/posts/${id}`;
+    return this.http.get(url, { responseType: 'text' });
+  }
+}
Index: blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.scss
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.scss b/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.scss
new file mode 100644
--- /dev/null	(date 1718540312735)
+++ b/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.scss	(date 1718540312735)
@@ -0,0 +1,62 @@
+
+.spinner{
+  height: 500px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.pagination {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  margin-top: 10px;
+
+  button {
+    background: none;
+    border: none;
+    cursor: pointer;
+    display: flex;
+    align-items: center;
+    margin: 0 10px;
+    padding: 5px;
+    font-size: 16px;
+    color: #0f0092;
+
+
+    &:disabled {
+      cursor: not-allowed;
+      color: #cccccc;
+    }
+
+    mat-icon {
+      font-size: 24px;
+    }
+  }
+
+  span {
+    margin: 0 10px;
+    font-size: 16px;
+  }
+}
+::ng-deep .check-btn {
+  background-color: #0f0092;
+  border: none;
+  color: white;
+  display: inline-block;
+  font-size: 14px;
+  cursor: pointer;
+  border-radius: 4px;
+  height: 20px;         /* Set the button height */
+  line-height: 20px;    /* Ensure the text is vertically centered */
+  padding: 0 5px;       /* Adjust padding to fit the text */
+
+  &:hover {
+    background-color: #0f0092;
+  }
+
+  &:disabled {
+    background-color: #cccccc;
+    cursor: not-allowed;
+  }
+}
\ No newline at end of file
Index: blogs-analyzer-ui/src/app/dashboard/components/header/header.component.scss
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.scss b/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.scss
--- a/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.scss	(revision 346accf1d385fae931da0747f821591d0c770d3f)
+++ b/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.scss	(date 1718540261438)
@@ -13,12 +13,8 @@
 .main-content {
   display: flex;
   padding: 20px;
-  margin-left: 400px;
 }
 
-.heading {
-  margin-left: 350px;
-}
 .navigation {
   flex: 1;
   padding-right: 20px;
@@ -45,7 +41,5 @@
   padding: 10px 20px;
   font-size: 16px;
   margin-bottom: 10px;
-  width: 60%;
+  width: 40%;
 }
-
-
Index: blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.html b/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.html
new file mode 100644
--- /dev/null	(date 1718539266599)
+++ b/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.html	(date 1718539266599)
@@ -0,0 +1,33 @@
+<ng-container *ngIf="loading; else dataLoaded">
+    <div class="spinner mt-5" data-testid="mabl-bulk-shrink-loader">
+        <div class="spinner-border text-primary" role="status">
+            <span class="sr-only"></span>
+        </div>
+    </div>
+</ng-container>
+<ng-template #dataLoaded>
+    <ag-grid-angular
+            style="width: 100%; height: 500px;"
+            class="ag-theme-alpine"
+            [rowData]="rowData"
+            [columnDefs]="columnDefs"
+            [suppressDragLeaveHidesColumns]="true"
+            [enableCellTextSelection]="true"
+            [ensureDomOrder]="true"
+    ></ag-grid-angular>
+    <div class="pagination">
+        <button (click)="navigateToFirstPage()" [disabled]="currentPage === 1">
+            <mat-icon>keyboard_double_arrow_left</mat-icon>
+        </button>
+        <button (click)="onPageChange(currentPage - 1)" [disabled]="currentPage === 1">
+            <mat-icon>navigate_before</mat-icon>
+        </button>
+        <span>Page {{currentPage}} of {{totalPages}}</span>
+        <button (click)="onPageChange(currentPage + 1)" [disabled]="currentPage === totalPages">
+            <mat-icon>navigate_next</mat-icon>
+        </button>
+        <button (click)="navigateToLastPage()" [disabled]="currentPage === totalPages">
+            <mat-icon>keyboard_double_arrow_right</mat-icon>
+        </button>
+    </div>
+</ng-template>
Index: blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.spec.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.spec.ts b/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.spec.ts
new file mode 100644
--- /dev/null	(date 1718372937200)
+++ b/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.spec.ts	(date 1718372937200)
@@ -0,0 +1,21 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { TabularViewComponent } from './tabular-view.component';
+
+describe('TabularViewComponent', () => {
+  let component: TabularViewComponent;
+  let fixture: ComponentFixture<TabularViewComponent>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [TabularViewComponent]
+    });
+    fixture = TestBed.createComponent(TabularViewComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
Index: blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.ts b/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.ts
new file mode 100644
--- /dev/null	(date 1718543007160)
+++ b/blogs-analyzer-ui/src/app/dashboard/components/tabular-view/tabular-view.component.ts	(date 1718543007160)
@@ -0,0 +1,89 @@
+import {Component, OnInit} from '@angular/core';
+import {BlogService} from '../../services/blog.service';
+
+@Component({
+    selector: 'app-tabular-view',
+    templateUrl: './tabular-view.component.html',
+    styleUrls: ['./tabular-view.component.scss']
+})
+export class TabularViewComponent implements OnInit {
+    protected columnDefs: any[];
+    protected rowData: any[];
+    loading: boolean = true;
+    currentPage: number = 1;
+    pageSize: number = 10;
+    isLastPage: boolean = false;
+    totalPages: number = 1;
+
+    constructor(private blogService: BlogService) {
+        this.columnDefs = [
+            {headerName: 'ID', field: 'id', width: 100},
+            {headerName: 'Title', field: 'title.rendered', flex: 2},
+            {
+                headerName: 'URL',
+                field: 'url',
+                flex: 1,
+                cellRenderer: function (params: { value: string; }) {
+                    return '<mat-icon class="material-icons visibility-icon" style="cursor: pointer; color: #0f0092">visibility</mat-icon>';
+                },
+                onCellClicked: function (params: { data: { url: string; }; }) {
+                    window.open(params.data.url, '_blank');
+                },
+            },
+            {headerName: 'Status', field: 'status', flex: 1},
+            {
+                headerName: 'Quality Check',
+                field: 'qualityCheck',
+                flex: 1,
+                cellRenderer: function (params: { data: { qualityCheck: string; }; }) {
+                    return '<button class="check-btn">Quality Check</button>';
+                }
+            }
+        ];
+
+        this.rowData = [];
+    }
+
+    ngOnInit(): void {
+        this.fetchPosts(this.currentPage);
+    }
+
+    fetchPosts(page: number): void {
+        this.loading = true;
+        this.blogService.getAllPosts(page, this.pageSize).subscribe(
+            (data) => {
+                this.rowData = data.posts.map((post: { id: any; title: any; url: any; status: any; }) => {
+                    return {
+                        id: post.id,
+                        title: post.title,
+                        url: post.url,
+                        status: post.status
+                    };
+                });
+                this.totalPages = data.totalPages;
+                this.isLastPage = data.isLastPage;
+                this.loading = false;
+            },
+            (error) => {
+                console.error('Error fetching posts:', error);
+                this.loading = false;
+            }
+        );
+    }
+
+    onPageChange(page: number): void {
+        if (page < 1 || (this.isLastPage && page > this.currentPage)) {
+            return;
+        }
+        this.currentPage = page;
+        this.fetchPosts(this.currentPage);
+    }
+
+    navigateToFirstPage(): void {
+        this.onPageChange(1);
+    }
+
+    navigateToLastPage(): void {
+        this.onPageChange(this.totalPages);
+    }
+}
Index: blogs-analyzer-ui/src/app/dashboard/components/header/header.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.ts b/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.ts
--- a/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.ts	(revision 346accf1d385fae931da0747f821591d0c770d3f)
+++ b/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.ts	(date 1718540618117)
@@ -1,4 +1,5 @@
 import { Component } from '@angular/core';
+import {BlogService} from "../../services/blog.service";
 
 @Component({
   selector: 'app-header',
Index: blogs-analyzer-ui/src/app/dashboard/components/header/header.component.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.html b/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.html
--- a/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.html	(revision 346accf1d385fae931da0747f821591d0c770d3f)
+++ b/blogs-analyzer-ui/src/app/dashboard/components/header/header.component.html	(date 1718540261442)
@@ -1,13 +1,6 @@
 <div class="header">
-  <img src="assets/images/nashtechlogo.png" alt="Nash Tech logo in red on a white background">
-  <h1>Blogs Analyzer</h1>
-  <img src="assets/images/home-button.svg" alt="Home icon in a house shape, orange color">
+    <img src="assets/images/nashtechlogo.png" alt="Nash Tech logo in red on a white background">
+    <h1>Blogs Analyzer</h1>
+    <img src="assets/images/home-button.svg" alt="Home icon in a house shape, orange color">
 </div>
-<div class="main-content">
-  <div class="search-section">
-    <h2 class = "heading">Search Blogs</h2>
-    <button class="search-button">Search by title</button>
-    <button class="search-button">Search by Name</button>
-    <button class="search-button">Search by Blog Id</button>
-  </div>
-</div>
+<app-home></app-home>
Index: blogs-analyzer-ui/src/styles.scss
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/styles.scss b/blogs-analyzer-ui/src/styles.scss
--- a/blogs-analyzer-ui/src/styles.scss	(revision 346accf1d385fae931da0747f821591d0c770d3f)
+++ b/blogs-analyzer-ui/src/styles.scss	(date 1718539805056)
@@ -1,1 +1,53 @@
 /* You can add global styles to this file, and also import other style files */
+@import 'ag-grid-community/styles/ag-grid.css';
+@import 'ag-grid-community/styles/ag-theme-alpine.css';
+
+html, body { height: 100%; }
+body {
+  margin: 0;
+  font-family: Arial;
+}
+
+.ag-theme-alpine {
+  --ag-control-panel-background-color: rgb(247, 243, 244);
+  --ag-selected-tab-underline-width: 2px;
+  --ag-selected-tab-underline-transition-speed: 0.5s;
+  --ag-header-background-color: #0f0092;
+  --ag-header-foreground-color: #f3f3f3;
+  --ag-border-color: transparent;
+}
+
+.ag-root-wrapper {
+  border: none;
+}
+
+.ag-header-row {
+  font-family: Arial;
+  font-size: 15px;
+  font-weight: 600;
+  line-height: 24px;
+}
+
+.ag-header-row:not(:first-child) .ag-header-cell:not(.ag-header-span-height.ag-header-span-total), .ag-header-row:not(:first-child) .ag-header-group-cell.ag-header-group-cell-with-group {
+  border-top: 1px dot-dot-dash #d1d1d1;
+}
+
+.ag-header{
+  border-bottom: 1px dotted #d1d1d1;
+}
+
+.ag-row {
+  height: 40px;
+  background-color: #fff;
+  font-family: Arial;
+  font-size: 14px;
+  font-weight: 400;
+  line-height: 17px;
+  border-bottom: 1px dotted #d1d1d1;
+  color: #666666;
+}
+
+.ag-theme-alpine .ag-row, .ag-theme-alpine-dark .ag-row {
+  font-size: 15px;
+}
+
Index: blogs-analyzer-ui/package.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/package.json b/blogs-analyzer-ui/package.json
--- a/blogs-analyzer-ui/package.json	(revision 346accf1d385fae931da0747f821591d0c770d3f)
+++ b/blogs-analyzer-ui/package.json	(date 1718372872905)
@@ -16,10 +16,12 @@
     "@angular/compiler": "^16.0.0",
     "@angular/core": "^16.0.0",
     "@angular/forms": "^16.0.0",
+    "@angular/material": "^16.1.1",
     "@angular/platform-browser": "^16.0.0",
     "@angular/platform-browser-dynamic": "^16.0.0",
     "@angular/router": "^16.0.0",
-    "@angular/material": "^16.1.1",
+    "ag-grid-angular": "^31.3.2",
+    "ag-grid-community": "^31.3.2",
     "bootstrap": "^5.3.0",
     "rxjs": "~7.8.0",
     "sonarqube-scanner": "^3.0.1",
Index: blogs-analyzer/src/main/java/com/nashtech/blogs/analyzer/controller/WordPressController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer/src/main/java/com/nashtech/blogs/analyzer/controller/WordPressController.java b/blogs-analyzer/src/main/java/com/nashtech/blogs/analyzer/controller/WordPressController.java
--- a/blogs-analyzer/src/main/java/com/nashtech/blogs/analyzer/controller/WordPressController.java	(revision 346accf1d385fae931da0747f821591d0c770d3f)
+++ b/blogs-analyzer/src/main/java/com/nashtech/blogs/analyzer/controller/WordPressController.java	(date 1718523637693)
@@ -5,14 +5,13 @@
 import org.json.JSONObject;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpMethod;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.client.RestTemplate;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 
 @RestController
 @RequestMapping("/api/wordpress")
@@ -74,8 +73,11 @@
 //    }
 
     @GetMapping
-    public ResponseEntity<List<Post>> getAllPosts() {
-        String url = WORDPRESS_API_BASE_URL + "posts";
+    public ResponseEntity<Map<String, Object>> getAllPosts(
+            @RequestParam(value = "page", defaultValue = "1") int page,
+            @RequestParam(value = "size", defaultValue = "10") int size
+    ) {
+        String url = WORDPRESS_API_BASE_URL + "posts?page=" + page + "&per_page=" + size;
         ResponseEntity<List<Map<String, Object>>> response = restTemplate.exchange(
                 url,
                 HttpMethod.GET,
@@ -97,13 +99,22 @@
                 title.setRendered((String) titleMap.get("rendered"));
                 post.setTitle(title);
 
-                post.setUrl(WORDPRESS_API_BASE_URL + "posts/" + post.getId());
+                String postUrl = (String) postMap.getOrDefault("link", ((Map<String, Object>) postMap.get("guid")).get("rendered"));
+                post.setUrl(postUrl);
                 post.setStatus((String) postMap.get("status"));
 
                 posts.add(post);
             }
         }
 
-        return ResponseEntity.ok(posts);
+        HttpHeaders headers = response.getHeaders();
+        int totalPosts = Integer.parseInt(Objects.requireNonNull(headers.getFirst("X-WP-Total")));
+        int totalPages = (int) Math.ceil((double) totalPosts / size);
+
+        Map<String, Object> responseBody = new HashMap<>();
+        responseBody.put("posts", posts);
+        responseBody.put("totalPages", totalPages);
+
+        return ResponseEntity.ok(responseBody);
     }
 }
Index: blogs-analyzer-ui/src/app/dashboard/components/home/home.component.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.html b/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.html
new file mode 100644
--- /dev/null	(date 1718540722414)
+++ b/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.html	(date 1718540722414)
@@ -0,0 +1,23 @@
+<div class="main-content">
+    <div class="search-section">
+        <div class="d-flex justify-content-around align-items-center m-3">
+            <div class="subscribe">
+                <p>Search Blogs by Text</p>
+                <input placeholder="Blog Title" class="subscribe-input" type="text" [(ngModel)]="searchTitle">
+                <br>
+                <button class="search-btn" (click)="searchByTitle()">Search</button>
+            </div>
+            <div class="subscribe">
+                <p>Search Blogs by ID</p>
+                <input placeholder="Blog Id" class="subscribe-input" type="number" [(ngModel)]="blogId">
+                <br>
+                <button class="search-btn" (click)="getBlogById()">Search</button>
+            </div>
+        </div>
+        <h2 class="mx-4">Existing Blogs</h2>
+
+        <mat-card class="px-2 py-1 mx-4" >
+            <app-tabular-view></app-tabular-view>
+        </mat-card>
+    </div>
+</div>
\ No newline at end of file
Index: blogs-analyzer-ui/src/index.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/index.html b/blogs-analyzer-ui/src/index.html
--- a/blogs-analyzer-ui/src/index.html	(revision 346accf1d385fae931da0747f821591d0c770d3f)
+++ b/blogs-analyzer-ui/src/index.html	(date 1718438819263)
@@ -6,6 +6,7 @@
   <base href="/">
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="icon" type="image/x-icon" href="favicon.ico">
+    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
 </head>
 <body>
   <app-root></app-root>
Index: blogs-analyzer-ui/src/app/dashboard/components/home/home.component.scss
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.scss b/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.scss
new file mode 100644
--- /dev/null	(date 1718540618129)
+++ b/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.scss	(date 1718540618129)
@@ -0,0 +1,88 @@
+.subscribe {
+  position: relative;
+  height: 140px;
+  width: 300px;
+  padding: 10px;
+  background-color: #FFF;
+  border-radius: 8px;
+  color: #333;
+  box-shadow: 0 0 10px 0 #9b9b9b,
+  0 1px 1px 0 #9b9b9b,
+  0 1px 1px 0 #9b9b9b;}
+
+.subscribe:after {
+  position: absolute;
+  content: "";
+  right: -10px;
+  bottom: 18px;
+  width: 0;
+  height: 0;
+  border-left: 0 solid transparent;
+  border-right: 10px solid transparent;
+  border-bottom: 10px solid #1a044e;
+}
+
+.subscribe p {
+  text-align: center;
+  font-size: 20px;
+  font-weight: bold;
+  letter-spacing: 4px;
+  line-height: 28px;
+}
+
+.subscribe input {
+  position: absolute;
+  bottom: 30px;
+  border: none;
+  border-bottom: 1px solid #d4d4d4;
+  padding: 10px;
+  width: 82%;
+  background: transparent;
+  transition: all .25s ease;
+}
+
+.subscribe input:focus {
+  outline: none;
+  border-bottom: 1px solid #0d095e;
+  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', 'sans-serif';
+}
+
+.subscribe .search-btn {
+  fill: transparent;
+  border: transparent;
+  position: absolute;
+  border-radius: 30px 0 0 30px;
+  background-color: #0f0092;
+  color: #FFF;
+  padding: 12px 25px;
+  display: inline-block;
+  font-size: 12px;
+  font-weight: bold;
+  letter-spacing: 5px;
+  right: -10px;
+  bottom: -20px;
+  cursor: pointer;
+  transition: all .25s ease;
+  box-shadow: -5px 6px 20px 0px rgba(26, 26, 26, 0.4);
+}
+
+.subscribe .search-btn:hover {
+  background-color: #07013d;
+  box-shadow: -5px 6px 20px 0px rgba(88, 88, 88, 0.569);
+}
+
+.tabular-view-wrapper {
+  display: inline;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+}
+
+.mat-mdc-card {
+  border-radius: 8px;
+  max-height: 800px;
+  min-height: 600px;
+  box-shadow: 0px 0px 10px 0px #9b9b9b,
+  0 1px 1px 0 #9b9b9b,
+  0 1px 1px 0 #9b9b9b;
+}
Index: blogs-analyzer-ui/src/app/dashboard/components/home/home.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.ts b/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.ts
new file mode 100644
--- /dev/null	(date 1718540618109)
+++ b/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.ts	(date 1718540618109)
@@ -0,0 +1,49 @@
+import { Component } from '@angular/core';
+import {BlogService} from "../../services/blog.service";
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.scss']
+})
+export class HomeComponent {
+  searchTitle!: string;
+  blogId!: number;
+
+  constructor(private blogService: BlogService) { }
+
+  ngOnInit(): void {
+  }
+
+  searchByTitle() {
+    if (this.searchTitle) {
+      this.blogService.searchPostsByTitle(this.searchTitle).subscribe(
+          (data) => {
+            const newTab = window.open();
+            if (newTab) {
+              newTab.document.body.innerHTML = data;
+            }
+          },
+          (error) => {
+            console.error('Error fetching posts by title:', error);
+          }
+      );
+    }
+  }
+
+  getBlogById() {
+    if (this.blogId) {
+      this.blogService.getPostById(this.blogId).subscribe(
+          (data) => {
+            const newTab = window.open();
+            if (newTab) {
+              newTab.document.body.innerHTML = data;
+            }
+          },
+          (error) => {
+            console.error('Error fetching post by ID:', error);
+          }
+      );
+    }
+  }
+}
Index: blogs-analyzer-ui/src/app/dashboard/components/home/home.component.spec.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.spec.ts b/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.spec.ts
new file mode 100644
--- /dev/null	(date 1718540173905)
+++ b/blogs-analyzer-ui/src/app/dashboard/components/home/home.component.spec.ts	(date 1718540173905)
@@ -0,0 +1,21 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { HomeComponent } from './home.component';
+
+describe('HomeComponent', () => {
+  let component: HomeComponent;
+  let fixture: ComponentFixture<HomeComponent>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [HomeComponent]
+    });
+    fixture = TestBed.createComponent(HomeComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
